restful.module

```diff
@@ -749,3 +768,21 @@ function restful_date_time_format_element_validate($element, &$form_state) {
     )));
   }
 }
+
+/**
+ * Determine if this is the first time we try to switch the user.
+ *
+ * @return bool
+ *   TRUE if this static function was already called.
+ */
+function restful_is_user_switched() {
```

Comment:
I agree in general to the statement, but for the sake of security, I think it would be ok to make the exception. If you feel strongly about it - I can change